shell/probe/nocturnal.sh

```diff
@@ -1,30 +1,33 @@
 #!/bin/bash
 
+PRELUDE_SLEEP=${PRELUDE_SLEEP:=14440}
+PRELUDE_DIR=${PRELUDE_DIR:="./cache"}
 PRELUDE_API=${PRELUDE_API:="https://api.preludesecurity.com"}
+PRELUDE_CA=${PRELUDE_CA:="prelude-account-prod-us-west-1.s3.amazonaws.com"}
 
-sys=$(uname -s)-$(uname -m)
-dos=$(echo $sys | tr '[:upper:]' '[:lower:]')
+dos=$(uname -s)-$(uname -m)
 
 while :
-do
-    temp=$(mktemp)
-    location=$(curl -sL -w %{url_effective} -o $temp -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
+do  
+    exe=$PRELUDE_DIR/$(uuidgen)
+    location=$(curl -sL -w %{url_effective} --create-dirs -o $exe -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
```

Comment:
will you make the curl flags include `curl -sfSL` so it'll print out any http error codes

Reply:
hmm, where would those print to in that case, as we're mapping to $location?

Reply:
the -f has the  http failure code print to stdout. location is still filled with url_effective

the -S isn't as necessary now that we're deleting the files, but that stop the file from being created on http error code. The --create-dirs still takes effect tho